src/nearest_neighbors/datasets/synthetic_data/loader.py

```diff
@@ -186,8 +187,90 @@ def _make_mcar(self) -> None:
         self.availability_mask = A
 
     def _make_mnar(self) -> None:
-        raise NotImplementedError("MNAR yet to be implemented")
-        # TODO: Aashish/Caleb/Kyuesong/Tatha: come back to this later
+        """Makes values missing not at random (MNAR), specifically staggered adoption (non-positivity & confounded)"""
+        U = self.row_latent
+
+        N = self.num_rows
+        T = self.num_cols
+
+        missing_mask = np.zeros((N, T))
+        pre_Masking = np.zeros((N, T))
+
+        # Divide units into 3 groups
+        g1_inds = np.arange(0, N // 3)
+        g2_inds = np.arange(N // 3, 2 * N // 3)
+        g3_inds = np.arange(2 * N // 3, N)
+
+        gamma_1 = [2, 0.7, 1, 0.7]
+        gamma_2 = [2, 0.2, 1, 0.2]
```

Comment:
What is the significance of γ1, γ2? These appear as [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)). Is there a way we can parametrize this? Should also put some documentation (inline or at the top) to explain how this entire function works.